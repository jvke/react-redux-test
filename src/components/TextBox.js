import React from 'react';
import styled from 'styled-components';

const TextBox = (props) => (
  <input {...props} type="text" />
);

export default styled(TextBox)`
  &:focus {
    outline: none;
  }
`;
