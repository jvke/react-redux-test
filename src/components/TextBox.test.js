import React from 'react';
import ReactDOM from 'react-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import TextBox from './TextBox';

describe('TextBox', () => {
  it('Textbox renders with no props specified', () => {
    const tree = renderer
      .create(<TextBox />)
      .toJSON();
    expect(tree).toHaveStyleRule('outline', 'none', { modifier: ':focus' })
    expect(tree).toMatchSnapshot();
  });

  it('Textbox renders with props specified', () => {
    const tree = renderer
      .create(<TextBox a={1} b={2} c={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
