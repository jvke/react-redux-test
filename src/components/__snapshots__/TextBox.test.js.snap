// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox Textbox renders with no props specified 1`] = `
.c0:focus {
  outline: none;
}

<input
  className="c0"
  type="text"
/>
`;

exports[`TextBox Textbox renders with props specified 1`] = `
.c0:focus {
  outline: none;
}

<input
  a={1}
  b={2}
  c={3}
  className="c0"
  type="text"
/>
`;
